tidy3d/components/mode/data/sim_data.py

```diff
@@ -0,0 +1,119 @@
+"""Mode solver simulation data"""
+
+from __future__ import annotations
+
+from typing import Literal, Tuple
+
+import pydantic.v1 as pd
+
+from ...base import cached_property
+from ...data.monitor_data import ModeSolverData
+from ...data.sim_data import AbstractYeeGridSimulationData
+from ...types import Ax, FieldVal, PlotScale
+from ..simulation import ModeSimulation
+
+# Warning for field intensity at edges over total field intensity larger than this value
+FIELD_DECAY_CUTOFF = 1e-2
+
+
+class ModeSimulationData(AbstractYeeGridSimulationData):
+    """Data associated with a mode solver simulation."""
+
+    simulation: ModeSimulation = pd.Field(
+        ..., title="Mode simulation", description="Mode simulation associated with this data."
+    )
+
+    data_raw: ModeSolverData = pd.Field(
+        ...,
+        title="Raw Data",
+        description="Raw mode data containing the field and effective index " "on unexpanded grid.",
+    )
+
+    data: Tuple[Literal[None], ...] = pd.Field(
+        (),
+        title="Monitor Data",
+        description="List of :class:`AbstractMonitorData` instances "
+        "associated with the monitors of the original :class:`AbstractSimulation`.",
+    )
+
+    @cached_property
+    def modes(self) -> ModeSolverData:
+        """:class:`.ModeSolverData` containing the field and effective index data.
+
+        Returns
+        -------
+        ModeSolverData
+            :class:`.ModeSolverData` object containing the effective index and mode fields.
+        """
+        mode_solver_data = self.data_raw
+        return mode_solver_data.symmetry_expanded_copy
+
+    def plot_modes(
```

Comment:
should we also provide a `SimulationData.plot_fields()` wrapper for this? in case people want to use the same api call?

Reply:
yeah. the only difference is that since there aren't any monitors, you don't pass a monitor name